packages/project-access/src/file/fileAccess.ts

```diff
@@ -0,0 +1,36 @@
+import { promises as fs } from 'fs';
```

Comment:
Do we have a use for these helper functions? We use `mem-fs` everywhere in the `open-ux-tools` we are not going directly to the filesystem.

Reply:
I removed the file access functions from top level export, but still need it internally, e.g. to find all apps in a given list of workspace root files.

For other functions, we need to check whether they should offer to read via `mem-fs`. Currently exported functions:

```
findAllApps()
getCapModelAndServices()
getWebappPath() 
loadModuleFromProject()
```

out of this list, these functions access the file system:
`findAllApps()`: accesses `package.json` and `manifest.json` to find apps 
`getWebappPath()`: accesses `ui5.yaml` to pass content to ui5-config


other candidates of functions, that access the file system, but are currently not exported are:
```
isCapProject()
isCapNodeJsProject()
isCapJavaProject()
```

Proposal to enable project access with mem-fs:
Once we have a proper app or project access structure in place, we can have a factory method that allows to pass optionally a `mem-fs` instance. All access methods on returned object would then use the `mem-fs` instance in case is was provided. E.g. in future something like

```typescript
function createProjectAccess(root:string, fs?: Editor): ProjectAccess
```

Reply:
@Klaus-Keller I gave `findFiles` and `findFileUp` a shot since these seem to be the lower level functions used for the functions above. My experiment is at https://github.com/SAP/open-ux-tools/pull/705/files. Opinions?

Reply:
The low level fs functions are supporting mem-fs. The optional mem-fs parameter needs to be exposed by its using functions as well.